gnd/src/main/res/layout/polygon_drawing_controls.xml

```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+  ~ Copyright 2021 Google LLC
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     https://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  xmlns:tools="http://schemas.android.com/tools">
+
+  <data>
+    <variable
+      name="viewModel"
+      type="com.google.android.gnd.ui.home.mapcontainer.MapContainerViewModel" />
+    <variable
+      name="homeScreenViewModel"
+      type="com.google.android.gnd.ui.home.HomeScreenViewModel" />
+  </data>
+
+  <androidx.constraintlayout.widget.ConstraintLayout
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:background="@color/colorGrey800">
+
+    <!-- TODO: Add highlighted icon overlay when showing feature sheet -->
+    <ImageView
+      android:id="@+id/map_crosshairs_img"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:adjustViewBounds="false"
+      android:src="@drawable/ic_crosshairs"
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent" />
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+      android:id="@+id/undo_button"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:layout_margin="16dp"
+      app:backgroundTint="@android:color/white"
+      app:fabSize="normal"
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintEnd_toStartOf="@+id/add_polygon_button"
+      app:srcCompat="@drawable/ic_undo_black"
+      app:tint="@android:color/black"
+      app:useCompatPadding="true" />
+
+    <Button
+      android:id="@+id/add_polygon_button"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:background="@android:color/white"
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintEnd_toEndOf="parent"
+      android:layout_marginBottom="32dp"
+      android:text="@string/add_point"
+      android:textColor="@color/colorAccent"
+      />
+
+    <Button
+      android:id="@+id/save_polygon_button"
```

Comment:
save polygon -> complete to match UI, here and below.

Reply:
updated